### DISCUSSION TITLE
Separate repository for web app

### DISCUSSION THREAD

**Discussion by owiecc** [typst/typst#1942] [General]
As web app and typst cli are effectively two separate software projects why not separate the repos? The web app repo can be empty and be used just to collect the UI/feature requests and bugs related to the web app.

**Comment by alerque** [comment]
I second this suggestion. As someone with only an interest in the CLI/renderer end of things I find it quite confusing to follow along with the stream of issues when half of them are web UI related and completely Orthogonal to the typesetting engine.

**Reply by Enivex** [reply]
For the moment, you can add `-label:"web app"` as a filter to entirely remove all the web app issues.

**Reply by alerque** [reply]
That only helps when browsing this repository directly. It does not help the notifications when you "watch" the repository and hence follow new issue activity. The labels do not even show up in the notifications page, one has to click through to the issue or guess from the title whether it is relevant or not

**Comment by laurmaedje** [comment]
Our reason for not doing that was to keep things simpler for people reporting bugs. Maybe it would not be a problem to find the other repository for people that report bugs on GitHub in the first place (instead of via feedback form). But we also already have a backlog of issues now and I don't know a way to migrate them. I'm not sure whether the change would be good and also not whether it would be worth the migration pain, but I'm open for more discussion. Since all web app issues have a label, they should be quite easy to filter out though.

**Reply by alerque** [reply]
You can move issues between projects in the same GitHub org. This would allow you to move out all the current backlog of unrelated issues (with all their history) as well as move new issues that got opened in the wrong place later (which is inevitable to happen sometimes). An issue template with a note about using a different location if the bug is about the webapp would also help mitigate them being opened in the wrong place at all.

When moving issues the *old* location gets closed and locked but stays around as a stub pointing to the new issue location, so existing links and mentions are fine. The *new* location will be an incremental number for the issue in the new location with all the comments and status stuff copied over so progress can continue.

**Reply by laurmaedje** [reply]
Thanks, that's very helpful information. I will discuss the option of a second repository with @reknih.

**Reply by PgBiel** [reply]
I second @laurmaedje's concerns, noting that new users might not be able to properly distinguish a web app issue from a compiler issue (we've seen cases like that before), and to fix it here would just involve changing one or more labels.

Personally, I think it's fine as is, even though I understand the appeal for further organization. I think there are cases where doing further separation is much more needed, but for Typst it has been working out so far imo.

**Reply by alerque** [reply]
You already have to triage issues. Either you add a label or your click the "move to" and pick the other repository. It isn't hard either way.

The status quo *hasn't* been working out so well for those of us who wish to "watch" the repository and be notified of issues but we only care about one half of the equation. Being interested in the renderer I'm bombarded with notifications about the web app that I don't have much interest in.

**Comment by laurmaedje** [comment]
We've come to the conclusion that we're okay with making this change. However, I can't promise when we'll get to it since it's a bit of work to set things up and migrate everything.

**Comment by laurmaedje** [comment]
This is now done. Web app issues should now be reported at https://github.com/typst/webapp-issues and we've added cross-links in the issue templates to guide people to the right place.
